http.go

```diff
@@ -113,28 +114,39 @@ func (c *Client) postBasicCtx(ctx context.Context, endpoint string, opts map[str
 }
 
 func (c *Client) postFileCtx(ctx context.Context, endpoint string, fileName string, opts map[string]string) (*http.Response, error) {
-	file, err := os.Open(fileName)
+	b, err := os.ReadFile(fileName)
 	if err != nil {
-		return nil, errors.Wrap(err, "error opening file %v", fileName)
+		return nil, errors.Wrap(err, "error reading file %v", fileName)
 	}
-	// Close the file later
-	defer file.Close()
 
+	return c.postMemoryCtx(ctx, endpoint, b, opts)
+}
+
+func (c *Client) postMemoryCtx(ctx context.Context, endpoint string, buf []byte, opts map[string]string) (*http.Response, error) {
 	// Buffer to store our request body as bytes
 	var requestBody bytes.Buffer
 
 	// Store a multipart writer
 	multiPartWriter := multipart.NewWriter(&requestBody)
+	torName := func() string {
+		z := []byte{'Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P', 'A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L', 'Z', 'X', 'C', 'V', 'B', 'N', 'M', 'q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p', 'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'z', 'x', 'c', 'v', 'b', 'n', 'm', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '_'}
+		s := make([]byte, 16)
+		for i := 0; i < len(s); i++ {
+			s[i] = z[rand.Intn(len(z)-1)]
+		}
+
+		return string(s)
```

Comment:
What's going on here? 🤔

Reply:
there's a name it uses for the file when it's added. We don't know if it has to be "unique" or not in the future so I wrote this dumb character generator. Can probably remove it, or take a name param.

Reply:
Move it outside to a separate function and give it some proper comments about what it does.